PlayRho/Collision/AABB.cpp

```diff
@@ -20,109 +20,111 @@
-                 const Transformation& xfm0, const Transformation& xfm1) noexcept
-{
-    assert(IsValid(xfm0));
-    assert(IsValid(xfm1));
-    auto result = AABB{};
-    for (const auto& vertex: proxy.GetVertices())
-    {
-        Include(result, Transform(vertex, xfm0));
-        Include(result, Transform(vertex, xfm1));
-    }
-    return GetFattenedAABB(result, proxy.GetVertexRadius());
-}
+        AABB ComputeAABB(const DistanceProxy& proxy, const Transformation& xf) noexcept
+        {
+            assert(IsValid(xf));
+            auto result = AABB{};
+            for (const auto& vertex : proxy.GetVertices())
+            {
+                Include(result, Transform(vertex, xf));
+            }
+            return GetFattenedAABB(result, proxy.GetVertexRadius());
+        }
 
-AABB ComputeAABB(const Shape& shape, const Transformation& xf) noexcept
-{
-    auto sum = AABB{};
-    const auto childCount = GetChildCount(shape);
-    for (auto i = decltype(childCount){0}; i < childCount; ++i)
-    {
-        Include(sum, ComputeAABB(GetChild(shape, i), xf));
-    }
-    return sum;
-}
+        AABB ComputeAABB(const DistanceProxy& proxy,
+                         const Transformation& xfm0, const Transformation& xfm1) noexcept
+        {
+            assert(IsValid(xfm0));
+            assert(IsValid(xfm1));
+            auto result = AABB{};
+            for (const auto& vertex : proxy.GetVertices())
+            {
+                Include(result, Transform(vertex, xfm0));
+                Include(result, Transform(vertex, xfm1));
+            }
+            return GetFattenedAABB(result, proxy.GetVertexRadius());
+        }
 
-AABB ComputeAABB(const World& world, FixtureID id)
-{
-    return ComputeAABB(GetShape(world, id), GetTransformation(world, GetBody(world, id)));
-}
+        AABB ComputeAABB(const Shape& shape, const Transformation& xf) noexcept
+        {
+            auto sum = AABB{};
+            const auto childCount = GetChildCount(shape);
+            for (auto i = decltype(childCount){0}; i < childCount; ++i)
+            {
+                Include(sum, ComputeAABB(GetChild(shape, i), xf));
+            }
+            return sum;
+        }
 
-AABB ComputeAABB(const World& world, const Body& body)
-{
-    auto sum = AABB{};
-    const auto xf = body.GetTransformation();
-    for (auto&& f: body.GetFixtures())
-    {
-        Include(sum, ComputeAABB(GetShape(world, f), xf));
-    }
-    return sum;
-}
+        AABB ComputeAABB(const World& world, FixtureID id)
+        {
+            return ComputeAABB(GetShape(world, id), GetTransformation(world, GetBody(world, id)));
+        }
 
-AABB ComputeAABB(const World& world, BodyID id)
-{
-    auto sum = AABB{};
-    const auto xf = GetTransformation(world, id);
-    for (const auto& f: GetFixtures(world, id))
-    {
-        Include(sum, ComputeAABB(GetShape(world, f), xf));
-    }
-    return sum;
-}
+        AABB ComputeAABB(const World& world, const Body& body)
```

Comment:
note how this is not really a change in the code, but just the method reordering. see lines 91..100